packages/langium/src/grammar/langium-grammar-validator.ts

```diff
@@ -71,16 +71,14 @@ export class LangiumGrammarValidator {
         }
     }
 
-    checkFirstGrammarRule(grammar: ast.Grammar, accept: ValidationAcceptor): void {
-        const firstRule = getEntryRule(grammar);
-        if (firstRule) {
-            if (isDataTypeRule(firstRule)) {
-                accept('error', 'The entry rule cannot be a data type rule.', { node: firstRule, property: 'name' });
-            } else if (firstRule.fragment) {
-                accept('error', 'The entry rule cannot be a fragment.', { node: firstRule, property: 'name' });
-            }
-        } else {
+    checkEntryGrammarRule(grammar: ast.Grammar, accept: ValidationAcceptor): void {
+        const entryRules = grammar.rules.filter(e => ast.isParserRule(e) && e.entry) as ast.ParserRule[];
+        if (entryRules.length === 0) {
             accept('error', 'This grammar is missing an entry parser rule.', { node: grammar, property: 'name' });
```

Comment:
In case there is at least one parser rule, please assign this error to the first parser rule and add a code action to fix the problem by adding the `entry` keyword prefix. This will ease migration from Xtext.